src/doc/rustdoc/src/how-to-write-documentation.md

```diff
@@ -240,6 +241,21 @@ This will render as
 
 See the specification for the [task list extension] for more details.
 
+### Smart punctuation
```

Comment:
I found [this cmark test](https://github.com/commonmark/cmark/blob/master/test/smart_punct.txt) from the pulldown-cmark PR that added smart punctuation https://github.com/raphlinus/pulldown-cmark/pull/470/files#r480030864. Should I link to that? It might not be worth it, but I thought it was worth suggesting :)